### ISSUE TITLE
-o stdout commandline switch not working

### ISSUE THREAD

**Issue by fcahoon** [csound/csound#946]
I'm running on Debian unstable; I experience this error both with the Debian-packaged csound and one I built from the devel branch today. I'm running the following "hello world" CSD with the "-o stdout" command-line option:
```
<CsoundSynthesizer>
<CsInstruments>

sr = 48000
ksmps = 32
nchnls = 2
0dbfs = 1

instr 1
iFreq = p4
iAmpl = ampdbfs(p5)

aOut poscil3 iAmpl, iFreq
outs aOut, aOut
endin

</CsInstruments>
<CsScore>
i1 0 2 200 -1
</CsScore>
</CsoundSynthesizer>
```
Here is the output I'm seeing (this one from the csound I built with today's code):
```
forrcaho@vortex:~/csound$ ../github/cs6make/csound -o stdout simple.csd 
0dBFS level = 32768.0
--Csound version 6.11 (double samples) Apr 14 2018
[commit: 4689ff9c5cdeb04bd7f1e09fec41df6b03c02e06]
libsndfile-1.0.28
UnifiedCSD:  simple.csd
STARTING FILE
Creating orchestra
closing tag
Creating score
rtaudio: ALSA module enabled
rtmidi: ALSA Raw MIDI module enabled
Elapsed time at end of orchestra compile: real: 0.001s, CPU: 0.001s
sorting score ...
        ... done
Elapsed time at end of score sort: real: 0.001s, CPU: 0.001s
graphics suppressed, ascii substituted
0dBFS level = 1.0
orch now loaded
audio buffered in 256 sample-frame blocks
Output file type changed to IRCAM for use in pipe
sfinit: cannot open fd 1
Not a valid SNDFILE* pointer.
inactive allocs returned to freespace
end of score.              overall amps:  0.00000  0.00000
           overall samples out of range:        0        0
1 errors in performance
Elapsed time at end of performance: real: 0.002s, CPU: 0.001s
```

**Comment by jpffitch** [comment]
Works for me on opendSuSE.  Will try Debian Jessie

**Comment by jpffitch** [comment]
Looking at your output I cannot see why it is talking about a pipe.  I see
``
# ./csound -o stdout foo.csd > xxx
0dBFS level = 32768.0
--Csound version 6.11 (double samples) Apr 13 2018
[commit: none]
libsndfile-1.0.25
UnifiedCSD:  foo.csd
STARTING FILE
Creating orchestra
closing tag
Creating score
Elapsed time at end of orchestra compile: real: 0.035s, CPU: 0.007s
sorting score ...
        ... done
Elapsed time at end of score sort: real: 0.035s, CPU: 0.008s
graphics suppressed, ascii substituted
0dBFS level = 1.0
orch now loaded
audio buffered in 256 sample-frame blocks
writing 1024-byte blks of shorts to stdout (WAV)
SECTION 1:
new alloc for instr 1:
B  0.000 ..  2.000 T  2.000 TT  2.000 M:  0.89125  0.89125
Score finished in csoundPerform().
inactive allocs returned to freespace
end of score.              overall amps:  0.89125  0.89125
           overall samples out of range:        0        0
0 errors in performance
Elapsed time at end of performance: real: 0.167s, CPU: 0.091s
512 1024 sample blks of shorts written to stdout (WAV)
``

What file type were you trying to use?

**Comment by jpffitch** [comment]
More searching.  That message ("Output file type changed...") should only occur if the output file name starts with a | which I do not see in your example.  There is another route to that message but that too looks like in a pipe situation.  As it works for me I think either we need more details of your environment or the help of the Debian packager (Felipe Sateler I think)

**Comment by fcahoon** [comment]
Hmmm, would the options CMake selected when I built Csound offer any clues? Here's the CMakeCache.txt file from my build.
[CMakeCache.txt](https://github.com/csound/csound/files/1911117/CMakeCache.txt)

**Comment by jpffitch** [comment]
The problem is that in your example -o stdout the data is not going anywhere and te file descriptor is not seekable.  In my tries I send the stdout to a file so it works.  
I have improved the error reporting but I have two questions
1: how did it work before as the code has not changed since 2014?

2: what are you expecting to go with raw audio to stdout (the terminal by default)?

**Comment by fcahoon** [comment]
I did in fact expect it to spew to the terminal; I was originally trying to pipe the output into sox so I could set csound to a high sample rate and downsample.

This seems to be working now (still doesn't spew to the terminal, but it will pipe into sox) so it appears I was doing something wrong before. I did in fact see the same error before when using a pipe, but I can't reproduce it now. Thanks for looking.
